### PR TITLE
Websocket streams

### PR DESCRIPTION
**Pull request by mdobson**
Adding a websocket link that conforms to the multiplex websocket protocol detailed here:

https://docs.google.com/document/d/1-vkl6o5PjdgsPACXEo80-sF9gPkK9JNIGKxZ8G-2rtM

### CODE REVIEW

**Comment by mdobson** [comment]
Currently this branch will not allow you to subscribe to `_peer/connect` and `_peer/disconnect` events.

**Comment by mdobson** [comment]
Fixt ^^

File: lib/api_resources/root.js [hunk 1]
```diff
@@ -225,6 +225,10 @@ RootResource.prototype._renderRoot = function(env, next) {
         title: this.server._name,
         rel: [rels.server],
         href: env.helpers.url.path('/servers/' + encodeURI(this.server.id) )
+      },
+      {
+        rel: [rels.events],
+        href: env.helpers.url.path('/events').replace(/^http/, 'ws')
```
Comment (kevinswiber) [inline lib/api_resources/root.js]: There are a few things to watch out for when building these URLs, such as... is there a forwarded protocol?  Is this an HTTP client connection or a SPDY client connection?  See here:  https://github.com/zettajs/zetta/blob/3a7eb1d7dcb69e91dc5c3c10bf0271a8faefd42f/lib/api_formats/siren/server.siren.js#L10-L15

File: lib/stream_topic.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+  if (topicComponents.length < 3 && topicComponents.indexOf('**') === -1) {
+    return;
+  }
+
+  if (topicComponents.length === 3) {
+    topicComponents.unshift(null);
+  }
+
+  function checkForRegex(s) {
```
Comment (kevinswiber) [inline lib/stream_topic.js]: Should we auto-insert a beginning `^` and a trailing `$`?  I think this is what argo-url-router does for regular expressions in routes.  It's also what happens in smaller subsets of Regular Expression syntax, such as what's found in XML Schema.

**Comment by kevinswiber** [comment]
This is really cool stuff.  Is any of this functionality exposed to the Zetta runtime or is it all C2Z/Z2Z API?  To enable this inside apps and scouts, we could probably do something like:

```js
module.exports = function(server) {
  var stream = server.createReadStream('*/led/*/state', { limit: 100 });
  stream.on('data', function(msg) {
    // awesome!
  });
};
```

File: lib/api_resources/root.js (hunk 1 above)
Comment (mdobson) [inline lib/api_resources/root.js]: +1. Added a test case for this, and functionality to properly calculate.

File: lib/stream_topic.js (hunk 2 above)
Comment (mdobson) [inline lib/stream_topic.js]: +1 it's added in.

**Comment by AdamMagaluk** [comment]
@kevinswiber Currently no, this PR just focuses on the ws side of things. But yes that would be great in the app.

File: lib/api_resources/root.js (hunk 1 above)
Comment (mdobson) [inline lib/api_resources/root.js]: Talked with @AdamMagaluk thinking that we actually remove this link for SPDY responses entirely. Currently this feature is only really going to work over websockets.

**Pull request merged**